pkg/cli/cli.go

```diff
@@ -88,6 +95,14 @@ func (c CLI) Run() (int, error) {
 	}
 
 	for _, fileToValidate := range foundFiles {
+		// Pkl validation requires the 'pkl' binary to be present
+		if fileToValidate.FileType.Name == "pkl" {
+			if !isPklBinaryPresent() {
+				fmt.Printf("Warning: 'pkl' binary not found, file %s will be ignored.\n", fileToValidate.Path)
+				continue
+			}
+		}
+
```

Comment:
I would move this in the pkl validator

Reply:
Same response as below

Reply:
Let's go back to the reason I raised the point of doing something else that what is in this PR

I see a problem in having a loop, then adding a dirty if hack in the loop.

The validator should be in charge of validating, it's not the loop that should do a big `if`.